week0002/mattfanz/spiral.php

```diff
@@ -0,0 +1,72 @@
+<?php
+// FUNCTION
+function checkValidInput($input)
+{
+    if (!is_numeric($input) || $input <= 0) {
+        fwrite(STDERR, "The input should be numberic and > 0".PHP_EOL);
+        exit;
+    }
+}
+
+// MAIN
+$handleIn = fopen("php://stdin", "r");
+$handleOut = fopen('php://stdout', 'w');
+
+fwrite($handleOut, "Input the number of rows N:");
```

Comment:
Ok so N is the number of rows or M is the number of rows? You need number of rows & number of columns.